numpy/core/src/multiarray/item_selection.c

```diff
@@ -2124,6 +2125,46 @@ count_nonzero_bytes_384(const npy_uint64 * w)
     return r;
 }
 
+#if NPY_SIMD
+static NPY_INLINE npy_intp
+count_nonzero_bytes(const npy_uint8 *d, npy_uintp unrollx)
+{
+    int count = 0;
+    int i = 0;
```

Comment:
What if the array has more than 2**31 elements?

Reply:
an overflow error will raised before calling this method, because the maximum value of input parameter  `shape[0]` is 2**31-1.

Reply:
Isn't `shape[0]` a `npy_intp` with maximum value 2**63-1?

Reply:
Sorry, my bad, The size of `npy_intp` is 8 bytes.